File: Pemrograman Lanjut/Modul 1/Tugas2.java
import java.util.Scanner;

public class Tugas2{
	
	public static void main(String[] args) {
		Scanner Input = new Scanner(System.in);
	
		String NIM , tempNIM;
		String Kode = "10370311" ;
		
		
		
		do{
			
			System.out.print("Masukan NIM Anda : ");
			NIM = Input.nextLine();
			int n = NIM.length()  ;
			
			if (n<15){
				System.out.println("NIM Anda kurang dari 15 digit ");
			
			}
			else if (n>15) { 
			
				System.out.println("NIM Anda lebih dari 15 digit ");
			
			}
			else{
				tempNIM  = NIM.substring(4, 12);
				System.out.println(tempNIM) ;
				if(tempNIM.equals(Kode)){
			
					System.out.println("NIM "+NIM +" Adalah Mahasiswa Informatika ");
					break ;
				
				}else{
					
					System.out.println("NIM "+NIM +" Adalah BUKAN Mahasiswa INFORMATIKA ");
				}
			}
			
		} while (9>0) ;
  
   }
	 
}

File: Pemrograman Lanjut/Modul 1/Tugas 2.c
#include <stdio.h>
#include <string.h>

char NIM[50],tempNIM[12],Kode[9]={'1','0','3','7','0','3','1','1'};
int cek=0;

int main(){

   do{
   	  printf("Masukan NIM Anda : ");    
  	  scanf("%s", NIM);
   	
   if(strlen(NIM) < 15)
    {
        printf("NIM yang anda masukan kurang dari 15 angka\n");
        continue ;
    }
    else if(strlen(NIM) > 15)
    {
        printf("NIM yang anda masukan lebih dari 15 angka\n");
        continue ;
    }
    
	strncpy(tempNIM,&NIM[4],8);
	printf("%s",tempNIM);
    cek = strcmp(Kode,tempNIM);
    
	if(cek==0){
        printf("NIM %s adalah Mahasiswa Informatika\n",NIM);
        break ;
    }
	else{
        printf("NIM yang dimasukkan Bukan NIM Informatika\n");
    }
    
	}while(9>0);
    
	
	return 0;
    }

